pkg/proxy/config/config.go

```diff
@@ -127,19 +128,22 @@ func (s *endpointsStore) Merge(source string, change interface{}) error {
 	case ADD:
 		glog.V(4).Infof("Adding new endpoint from source %s : %+v", source, update.Endpoints)
 		for _, value := range update.Endpoints {
-			endpoints[value.Name] = value
+			cachekey := types.NewNamespacedName(value.Namespace, value.Name).CacheKey().String()
+			endpoints[cachekey] = value
```

Comment:
No good reason. That was the result of a minimally disruptive merge against
your code :-)  I'll fix it once the rest of your comments are in.

On Mon, Mar 9, 2015 at 9:42 PM, Tim Hockin <notifications@github.com> wrote:

> In pkg/proxy/config/config.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5192#discussion_r26098541>
> :
>
> > @@ -127,19 +128,22 @@ func (s *endpointsStore) Merge(source string, change interface{}) error {
> >  	case ADD:
> >  		glog.V(4).Infof("Adding new endpoint from source %s : %+v", source, update.Endpoints)
> >  		for _, value := range update.Endpoints {
> > -			endpoints[value.Name] = value
> > +			cachekey := types.NewNamespacedName(value.Namespace, value.Name).CacheKey().String()
> > +			endpoints[cachekey] = value
>
> Why not key endpoints on NamespacedName rather than string?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/5192/files#r26098541>
> .
>